src/mixings/dirichlet_mixing.hpp

```diff
@@ -21,6 +21,7 @@ class DirichletMixing : public BaseMixing {
  public:
   struct State {
     double totalmass;
+    double logtotmass;
   };
```

Comment:
This I'm not a fan of. The state should not be composed of two equivalent values, IMO. Is there any way around it? It's not a big deal though, just stick with it if it can't be helped

Reply:
we can remove it from the state and have it as a protected member of DirichletMixing, would that be better?

Reply:
That would be great! IMO it is similar to tau0_inverse from NNW, it is useful to keep track of but it's just an internal utility which neither the user nor other classes should know anything about

Reply:
Actually I spotted a bug with tau0_inverse.
While the hyperparams get updated in all the Hierarchies since they are a shared_ptr, tau0_inverse is updated only in the hierarchy that actually performs the update of the hyperparameters.
I fixed it by adding tau0_inverse to the hyperparameters :)

Reply:
Right, nice catch. Turning it into a shared pointer, out of the hypers of course, would be enough, correct?

Reply:
Soo about this and tau0_inv...

Reply:
Given that this State and the Hyperparameters are nested structs, their use is to be confined within the class they are defined into. Hence, i would not make too much of a deal of storing "redundant" information as long as it is actually useful.
To pass stuff to the user and across different parts of the code, we always use protos where the "real" state is defined.

Reply:
Right, I can agree with that, it's for internal use only. Let's just make sure that we also always set the inverse when getting the state from a proto